mcs/db_instance_client.go

```diff
@@ -0,0 +1,764 @@
+func InstanceUpdateAutoExpand(client DatabaseClient, id string, opts OptsBuilder) (r instances.ActionResult) {
+	b, err := opts.Map()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	reqOpts := getRequestOpts(202)
+	var result *http.Response
+	result, r.Err = client.Patch(getURL(client, instancesAPIPath, id), &b, nil, reqOpts)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceDetachConfigurationGroup ...
+func InstanceDetachConfigurationGroup(client DatabaseClient, id string) (r ConfigurationResult) {
+	opts := InstanceDetachConfigurationGroupOpts{
+		Instance: map[string]interface{}{},
+	}
+	b, err := opts.Map()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	reqOpts := getRequestOpts(202)
+	var result *http.Response
+	result, r.Err = client.Put(getURL(client, instancesAPIPath, id), b, nil, reqOpts)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceAttachConfigurationGroup ...
+func InstanceAttachConfigurationGroup(client DatabaseClient, id string, opts OptsBuilder) (r ConfigurationResult) {
+	b, err := opts.Map()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	reqOpts := getRequestOpts(202)
+	var result *http.Response
+	result, r.Err = client.Put(getURL(client, instancesAPIPath, id), b, nil, reqOpts)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceAction ...
+func InstanceAction(client DatabaseClient, id string, opts OptsBuilder) (r ActionResult) {
+	b, err := opts.Map()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	var result *http.Response
+	result, r.Err = client.Post(instanceActionURL(client, instancesAPIPath, id), b, nil, nil)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceRootUserEnable ...
+func InstanceRootUserEnable(client DatabaseClient, id string, opts OptsBuilder) (r CreateRootUserResult) {
+	b, err := opts.Map()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	reqOpts := getRequestOpts(200)
+	var result *http.Response
+	result, r.Err = client.Post(rootUserURL(client, instancesAPIPath, id), b, &r.Body, reqOpts)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceRootUserGet ...
+func InstanceRootUserGet(client DatabaseClient, id string) (r IsRootUserEnabledResult) {
+	var result *http.Response
+	result, err := client.Get(rootUserURL(client, instancesAPIPath, id), &r.Body, nil)
+	if err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceRootUserDisable ...
+func InstanceRootUserDisable(client DatabaseClient, id string) (r DeleteRootUserResult) {
+	reqOpts := getRequestOpts(200)
+	var result *http.Response
+	result, r.Err = client.Delete(rootUserURL(client, instancesAPIPath, id), reqOpts)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// InstanceDelete ...
+func InstanceDelete(client DatabaseClient, id string) (r DeleteResult) {
+	var result *http.Response
+	result, r.Err = client.Delete(getURL(client, instancesAPIPath, id), nil)
```

Comment:
ALL delete requests should have retries for non successful code

Reply:
The same, lets do it in a separate pr.